optional.go

```diff
@@ -82,17 +82,28 @@ func (o Optional[T]) ElseZero() (value T) {
 // representation of the zero value of the type wrapped if there is no value
 // wrapped by this optional.
 func (o Optional[T]) String() string {
-	return fmt.Sprintf("%v", o.ElseZero())
+	if v, ok := o.Get(); ok {
```

Comment:
The comment for the function needs updating as it says if no value is set that the zero value of the type is returned, which is no longer the case.